File: algorithm/search/linear_search.c
#include <stdio.h>
#include <stdbool.h>
bool search(int arr[], int size, int key)
{
    for (int i = 0; i < size; i++)
    {
        if (arr[i] == key)
        {
            return true;
        }
    }
    return false;
}
void input(int arr[], int size)
{
    printf("Enter the elemet of the array: \n");
    for (int i = 0; i < size; i++)
    {
        scanf("%d", &arr[i]);
    }
}
void printArr(int arr[], int size)
{
    for (int i = 0; i < size; i++)
    {
        printf("%d\t", arr[i]);
    }
    printf("\n");
}
int main()
{
    int size;
    printf("Enter the size of the array: \n");
    scanf("%d", &size);
    int arr[size];
    input(arr, size);
    printArr(arr, size);
    printf("Enter the key what you want to find: ");
    int key;
    scanf("%d", &key);
    bool result = search(arr, size, key);
    if (result == false)
    {
        printf("Not found");
    }
    else
    {
        printf("Found");
    }
}

File: Java/Algorithms/search/LinearSearch.java
package Java.Algorithms.search;

public class LinearSearch {
    public int search(int[] arr,int n,int key){
        for(int i =0;i<n;i++){
            if(arr[i]==key){
                return i;
            }
        }
        return -1;
    }
    public static void main(String[] args) {
        int[] arr = {10,3,5,6,8,40};
        LinearSearch s1 = new LinearSearch();

        int result = s1.search(arr, arr.length, 6);
        if(result==-1){
            System.out.println("Key is not found");
        }else{
            System.out.println("Key is found at index "+result);
        }
    }
}
